clippy_lints/src/utils/hir_utils.rs

```diff
@@ -673,4 +673,100 @@ impl<'a, 'tcx: 'a> SpanlessHash<'a, 'tcx> {
             },
         }
     }
+
+    pub fn hash_lifetime(&mut self, lifetime: &Lifetime) {
+        if let LifetimeName::Param(ref name) = lifetime.name {
+            match name {
+                ParamName::Plain(ref ident) => {
+                    ident.name.hash(&mut self.s);
+                },
+                ParamName::Fresh(ref size) => {
+                    size.hash(&mut self.s);
+                },
+                _ => {},
+            }
+        }
+    }
+
+    pub fn hash_ty(&mut self, ty: &Ty) {
+        std::mem::discriminant(&ty.node).hash(&mut self.s);
+        match ty.node {
+            TyKind::Slice(ref ty) => {
```

Comment:
You should use just `ty::Slice(..)` here, instead of `TyKind::Slice(..)`

Reply:
Do you have a link to any docs for this. As I'm matching on an instance of `TyKind` I'm not sure what the benefits/difference is?

Reply:
There's no docs there yet. It is currently an unwritten convention. So nothing you could have known.

I'm currently working on rustc internal lints, where it will warn you on using TyKind in the future: rust-lang/rust#59316

Also related: https://github.com/rust-lang/rust-clippy/pull/3889#discussion_r266190082

Reply:
Does that mean I should also do the same for every match arm?

Reply:
Yeah, `TyKind` shouldn't be used at all.